### DISCUSSION TITLE
Helm: Adopt a more secure Flask SECRET_KEY by default

### DISCUSSION THREAD

**Discussion by cwegener** [apache/superset#23823] [Ideas]
On the heels of the pending action items from CVE-2023-27524, it looks like there could be a significant improvement in secure SECRET_KEY values by adopting some changes to the helm chart similar to what Airflow have implement in the past here: https://github.com/apache/airflow/pull/17142

I don't have much experience with helm and I'm curious to hear what more experienced helm users think of Airflow's implementation.

**Comment by rusackas** [comment]
Thanks for the thread! CC-ing a few folks for eyes and input here: 
@dpgaspar @mistercrunch @craig-rueda @villebro @nytai

**Reply by cwegener** [reply]
Sharing my motivation for this topic:

1. Looking through a quick shodan search the situation about public Superset instances and their use of default secret keys needs improving.
2. Comparing Superset and Airflow in a shodan search, shows that Airflow's secret key usage is much improved.
3. From "eye balling" the number of helm chart questions for Superset, it seems that helm is a popular deployment method

So, my hypothesis is that by adopting the improvement that Airflow have done in their helm chart, Superset can benefit with a "secure by default" installation option for helm deployments.

**Comment by dpgaspar** [comment]
it's an interesting approach, their automatically using a random key if a `SECRET_KEY` is not set, this random key is stored has a kubernetes secret so that it can be shared between web servers and workers, this way every pod uses the same secret.

Superset also depends on the secret key to encrypt fields on the database so changing the secret would require users to follow https://superset.apache.org/docs/installation/configuring-superset#secret_key-rotation to decrypt and re-encrypt all their sensitive metadata db fields.

Another more explicit approach would be to just let the flask app refuse to start if a default secret key is found (our current default behaviour outside of helm), or use helm's `required` function with an explicit error message, urging the user to set their own unique `SECRET_KEY`
